modules/collectors.coffee

```diff
@@ -3,16 +3,23 @@ _ = require 'underscore'
 
 load = require "../lib/load"
 modules = require("config").modules
+useWhitelistedKeys = require("config").useWhitelistedKeys
+whitelistedkeys = ''
 
 module.exports = ({app, logger, config}, next) ->
   collectorHandler = (collectors) ->
     return (req, res) ->
+      if useWhitelistedKeys
+        if not _.every(_.keys(req.body), (v) -> _.contains(whitelistedkeys, v))
+          res.send(406, '')
+          return
       res.send(204, '')
 
       for coll in collectors
         coll(req.body, {req, res})
 
   logger.log "Loading collectors: #{ modules.collectors.join(', ') }"
+  whitelistedkeys = "#{modules.whitelistedKeys}".split(',')
```

Comment:
The values are coming from the YAML parser as an array: https://github.com/HubSpot/BuckyServer/pull/22/files#diff-fe7044f2ecd69c76ce484ad03fabc12fR41

They will only become a comma separated list if you cast them into a string, as you're doing with the string interpolation.